irksome/base_time_stepper.py

```diff
@@ -26,7 +26,8 @@ def __init__(self, F, t, dt, u0,
             "dt": dt,
             "u0": u0,
             "bcs": bcs,
-            "nullspace": nullspace}
+            "nullspace": nullspace,
+            "stepper": self}
```

Comment:
This feels redundant - `stepper` already includes `F`, `t`, `dt`, `u0`, `nullspace`, and `bcs` as data, so we now stash those pieces twice.  Is there a good design reason to include things both directly and indirectly?

Reply:
So you discovered this trick, too.  The question is whether removing them fro the appctx would harm any users who are not politely obeying the public interface.